app/src/main/java/com/example/myjokes/ui/joke_list/JokesListActivity.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.myjokes.ui.joke_list
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.myjokes.data.Joke
+import com.example.myjokes.data.JokeGenerator
+import com.example.myjokes.databinding.ActivityJokesListBinding
+import com.example.myjokes.ui.joke_details.JokeDetailsActivity
+import com.example.myjokes.ui.joke_list.recycler.adapter.JokeAdapter
+import com.example.myjokes.ui.joke_list.recycler.adapter.JokeListAdapter
+import com.example.myjokes.ui.joke_list.recycler.util.JokeItemCallback
+
+class JokesListActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityJokesListBinding
+    private lateinit var viewModel: JokeListViewModel
+    val generator = JokeGenerator
+
+    private val adapter = JokeAdapter {
+        startActivity(JokeDetailsActivity.getInstance(this, it))
+    }
+
+//    private val adapter2 = JokeListAdapter (
+//        JokeItemCallback(),
+//        startActivity(JokeDetailsActivity.getInstance(this, it))
+//    ) - я если честно не понял почему мне пишет ошибку, что startActivity возвращает Unit, вместо (Int) -> Unit, а с JokeAdapter такого нет
```

Comment:
Ты пытаешься передать сам метод startActivity в параметры. startActivity имеет вид () -> Unit, тогда как у тебя ожидается метод (Int) -> Unit. Ты можешь обернуть { startActivity(JokeDetailsActivity.getInstance(this, it)) } тогда будет работать аналогично тому, как в первом адаптере. Ну или таким образом получится то же самое
```suggestion
    private val adapter2 = JokeListAdapter (JokeItemCallback()) {
        startActivity(JokeDetailsActivity.getInstance(this, it))
    }
```

Reply:
Прикольно, не думал даже что лямбду можно так вынести